crates/policy-evaluator/src/policy_evaluator.rs

```diff
@@ -105,4 +106,50 @@ impl PolicyEvaluator {
             }
         }
     }
+
+    pub fn validate_settings(&mut self) -> SettingsValidationResponse {
+        let settings_str = match serde_json::to_string(&self.settings) {
+            Ok(s) => s,
+            Err(e) => {
+                return SettingsValidationResponse {
+                    valid: false,
+                    message: Some(format!("Cannot serialize validation params: {}", e)),
+                }
+            }
+        };
+
+        match self
+            .wapc_host
+            .call("validate_settings", settings_str.as_bytes())
+        {
+            Ok(res) => {
+                let vr: Result<SettingsValidationResponse> = serde_json::from_slice(&res)
+                    .map_err(|e| anyhow!("cannot convert response: {:?}", e));
+                vr.unwrap_or_else(|e| SettingsValidationResponse {
+                    valid: false,
+                    message: Some(format!("error: {:?}", e)),
+                })
+            }
+            Err(err) => {
+                if let wapc::errors::ErrorKind::GuestCallFailure(m) = err.kind() {
+                    // Unfortunately waPC doesn't define a dedicated error
+                    if m.contains("No handler registered") {
```

Comment:
Given this message could change, could we check the Wasm exports instead in order to determine if there exists a `validate_settings` function?

Reply:
Yes, that's right. I'll try to do that, I've already played with wasm disassemblers "via code"

Reply:
As discussed, let's merge this PR because it does some code reorganization. I'll file a new issue to track this improvement